### ISSUE TITLE
Loss Implementation Problem

### ISSUE THREAD

**Issue by axhiao** [Tiny-Snow/IR-Benchmark#3]
Hi, 

I just noticed that the implementation of loss function in your code is a little bit different from the one in your paper. 

In your code, [here](https://github.com/Tiny-Snow/IR-Benchmark/blob/1dd8de9b6335b412b152aaf0e615e2bef3d6a95b/src/itemrec/optimizer/optim_PSL.py#L183) is your PSL implemetation:

```python
            loss = torch.logsumexp(self.sigma(d) / self.tau_star, dim=1).mean()
```
where there is an $\exp$ operator in the logsumexp. 

But in your paper, according to the equation (4.1), the loss is:

<img width="522" alt="Image" src="https://github.com/user-attachments/assets/a4b325ab-97fc-44a0-a67f-a28a1574c814" />

May I ask:

1. why you put an $\exp$ in your code? Is there any reason for this? 

2. why you use $\sigma(d_{uij}) / \tau ^ *$ in your code, instead of $\sigma(d_{uij})^{1/\tau}$ as shown in your paper? 

Look forward to your response. Thank you!

**Comment by Tiny-Snow** [comment]
Thanks for your question.

The implementation is essentially in line with the paper. As you can see in [optim_PSL.py #L141-L146](https://github.com/Tiny-Snow/IR-Benchmark/blob/main/src/itemrec/optimizer/optim_PSL.py#L141-L146), `self.sigma` is slightly different from the $\sigma(\cdot)$ in the original paper, as it is essentially $\log \sigma(\cdot)$. Thus, `torch.logsumexp(self.sigma)` is equivalent to the Equation (4.1), i.e., $\log \sum_{j} \sigma(d_{uij})^{1 / \tau}$.

This implementation is mainly due to considerations of numerical accuracy, as `torch.logsumexp` helps it.

Feel free to ask me if you have more questions!
